python/semantic_kernel/functions/kernel_function_decorator.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+
+from inspect import Parameter, signature
+from typing import Callable, ForwardRef, Tuple
+
+
+def kernel_function(
+    *,
+    description: str = "",
+    name: str = "",
+):
+    """
+    Decorator for kernel functions.
+
+    Args:
+        description -- The description of the function
+        name -- The name of the function
+        input_description -- The description of the input
+        input_default_value -- The default value of the input
+    """
+
+    def decorator(func: Callable):
+        func.__kernel_function__ = True
+        func.__kernel_function_description__ = description or func.__doc__ or ""
+        func.__kernel_function_name__ = name or func.__name__
+        func.__kernel_function_context_parameters__ = []
```

Comment:
I must be missing something. We still do need `kernel_function_context_parameters`?

Reply:
Yes, but we now fill that here instead of is a separate decorator